Classes/EventListener/Mapping/EventToListenerMapping.php

```diff
@@ -36,21 +36,27 @@ final class EventToListenerMapping implements \JsonSerializable
      */
     private $options;
 
-    private function __construct(string $eventClassName, string $listenerClassName, array $options)
+    private function __construct(string $eventClassNamePattern, string $listenerClassName, array $options)
     {
-        $this->eventClassName = $eventClassName;
+        $this->eventClassNamePattern = $eventClassNamePattern;
         $this->listenerClassName = $listenerClassName;
         $this->options = $options;
     }
 
-    public static function create(string $eventClassName, string $listenerClassName, array $options): self
+    public static function create(string $eventClassNamePattern, string $listenerClassName, array $options): self
     {
-        return new static($eventClassName, $listenerClassName, $options);
+        return new static($eventClassNamePattern, $listenerClassName, $options);
     }
 
-    public function getEventClassName(): string
+    public function getEventClassNamePattern(): string
     {
-        return $this->eventClassName;
+        return $this->eventClassNamePattern;
+    }
+
+    public function matchesEventClassName(string $eventClassName): bool
```

Comment:
Is full regex matching really required here? Do we not just need a simple "all" wildcard?